### ISSUE TITLE
Multi-adapter PEFT loading

### ISSUE THREAD

**Issue by lxe** [oobabooga/text-generation-webui#853]
**Description**

https://github.com/huggingface/peft/pull/263 just got merged into PEFT. I think we can take a stab ad multi-lora loading. I'm using it here: https://github.com/lxe/simple-llm-finetuner/blob/master/trainer.py#L59

**Additional Context**

**Comment by oobabooga** [comment]
Very nice! Thanks for the heads up. Cc @mcmonkey4eva

**Comment by mcmonkey4eva** [comment]
Woo!

Main question in my head is how to build the interface for multi-lora-loading?

Few options:

1: Literally a text line where you type model names, `Loras: lora_a, my_lora_b, etc`. Maybe some javascript hack to validate it as you type.

2: The same current dropdown, but when you select a new LoRA, it just applies over the top of the last one, and only resets if you hit `None`

3: Just list every LoRA as checkboxes. This is the worst option as it'd be hell to make gradio work with it and also it prevents ordering.

4: An interface like what Auto WebUI uses for _styles_:
![image](https://user-images.githubusercontent.com/4000772/230691953-5107668e-2650-487d-80e8-ecad4be59399.png)

Where it's a dropdown that you can select things in, and they get populated into a list of lil bubbles, and you can hit `X` to remove ones.

All of the above options other than `2` would require an `Apply` button or similar activation such that LoRAs only load and apply once the user is ready for it.
(Unless we can find a way to accelerate rapid load/unload of LoRAs? ie process them during inference rather than applying over the model? Might be dirty. Would cost VRAM and speed too, so not great.)

**Comment by oobabooga** [comment]
I think that we could have a `gr.CheckboxGroup` where the user can select/deselect the desired LoRAs.

https://gradio.app/docs/#checkboxgroup

![cg](https://user-images.githubusercontent.com/112222186/230694834-1223a1e9-8cc4-4e0a-b94a-7db85f2d0985.png)

In principle, we could parse the `adapter_config.json` for all available LoRAs and only show those that match the currently selected model, since there is a `base_model_name_or_path` field. The problem is that the model names are all over the place. Here are two examples:

```
"base_model_name_or_path": "zpn/llama-7b",
"base_model_name_or_path": "decapoda-research/llama-7b-hf",
```

**Comment by mcmonkey4eva** [comment]
Yknow, I was thinking LoRA order mattered and therefore options like that were insufficient, but in hindsight I'm not sure that order actually does matter.
Seems like because at the endpoint it maps down to simple multiplications, the order can be swapped without changing the outcome.

**Comment by mcmonkey4eva** [comment]
I submitted a PR for this #1103
